### DISCUSSION TITLE
How can you reset the track appearance to default?

### DISCUSSION THREAD

**Discussion by sonora** [osmandapp/OsmAnd#11560] [General]
With newer builds I encounter oddities when displaying some of the tracks in my big gpx folder and subfolders. (These tracks don't usually contain any formatting information, and their file date shows they have not changed in years. I mostly use the Touring view Map style.)

1. **Wrong default width:** Some (strangely not all) now default to being displayed with track width 1 point, i.e. barely visible. (Checking their Appearance/Width tab, it seems that 'Custom' with value 1 is in deed selected for these.But I had never set this.)

2. **Missing indication of currently used values:** I initially I believed they were displayed transparently, hence I changed their color. How can I reset them to display in the map style's gpx default color? We have the following bugs now: Checking a track __correctly displayed__ in e.g. Touring view's defaults (as specified in the renderer), that track's Appearance tab shows neither a color selected, nor a width selected. It seems that that dialog does not cope with color or width values which are not contained in its own preselections?

3. **Missing track appearance reset:** The new Appearance dialog is missing a reset or default button. Hence it is unclear how to reset any track's appearance so that it is again displayed with the renderer (or track-intrinsic) defaults.

3. **GPX database seems sticky:** Some of this may be due to history of nightly developer builds? (Although I seem to have these issues also on rather newly installed test devices). But we are missing a viable way to reset the gpx appearance database? (It seems that temporarily renaming my Tracks folder does not do the trick, all bogus track appearance information was back after renaming the folder back?)

**Acceptance criteria to fix this:**

a. When first displaying a track from the tracks folder which does not contain any appearance information, the renderer defaults must be used.

b. The track appearance dialog must in all cases indicate the corresponding color and width used.

c. That track appearance dialog must also provide a 'reset to defaults' button to return a track's appearance to its defaults

d. There must be a way to return **all** track appearance to their default, best via resetting the gpx database then this can also be used for cases where the gpx database became corrupt.

**Comment by vshcherb** [comment]
I think most valuable part is c. That track appearance dialog must also provide a 'reset to defaults'. Everything else is fixable / avoidable. GPX database memory could / should be fixed during reimport asking to restore appearance or not. Though if we have "Reset" button why would we bother then about it.
GPX database was especially introduced to keep original tracks back. So indeed this is partially implemented feature (not usable yet), though I don't agree we need to "revert" something if we didn't change something i.e. we didn't change original files.
I would summarize we need to still do 3 things:
1. Finish up redesign of "Display GPX Tracks" (Configure Map) - we didn't even started it since last year cause it has massive requests 
2. Add button restore original appearance to the track (probably most critical)
3. On Share button have a checkbox to share file with original appearance or with custom

**Comment by sonora** [comment]
Plus: Investigate why some tracks from their first display on are shown with width 1 now in newer builds, leading users to believe they are not displayed at all. That seems to be the 'bug' portion of my report.

All the rest can be viewed as an incomplete implementation of both a feature and the corresponding customization dialog, agreed.

**Reply by Chumva** [reply]
This happens after the first track displaying?

**Reply by sonora** [reply]
No, it (recenrly) happens for tracks I  have had in my tracks folder for a long time, but I make visible for the first time.

**Comment by sonora** [comment]
I agree with Victor's comment above "_I think most valuable part is c._"

So how can we get this back into a Milestone?

**Reply by vshcherb** [reply]
I think we can create 1 specific issue, especially once we reached an agreement
